docs/applications.rst

```diff
@@ -3,4 +3,167 @@ Applications
 ============
 
 Adaptive ensemble simulation methods are implemented in terms of data flow
-operations and control flow logic.
\ No newline at end of file
+operations and control flow logic.
+
+Examples follow.
+
+Adaptive Markov State Modeling
+==============================
+
+In this example, we use a Python data flow scripting interface to connect
+PyEmma, a simulator adapter, and some additional Python code to iteratively
+run and analyze groups of simulations until convergence.
+
+The main script body defines and executes the subgraph.
+There is obvious opportunity for abstraction, but it may be illustrative in the
+present case to mix implementation details with algorithmic structure,
+in part to stimulate discussion and understanding of packaging and component
+capabilities.
+
+One supporting module (:file:`analyze.py`) wraps PyEmma.
+
+Another supporting module (:file:`gromacsmd.py`) wraps GROMACS command line tools and gmxapi.
+
+Main script body::
+
+    from gmxapi import modify_input
+    from gmxapi import mdrun as simulate
+
+    from gromacsmd import collect_samples, gro_to_pdb, make_input
+
+    # Could start with a list of distinct confs, but here we use a single starting point.
+    # GROMACS MD model and method parameters.
+    starting_structure = 'input_conf.gro' # GROMACS structure file.
+    topology_file = 'input.top' # GROMACS topology input.
+    run_parameters = 'params.mdp' # GROMACS simulation parameters.
+
+    initial_simulation_input = make_input(
+        simulation_parameters=run_parameters,
+        topology=topology_file,
+        conformation=starting_structure)
+
+    # Set up an array of N simulations, starting from a single input.
+    initial_input = scalems.broadcast(initial_simulation_input, shape=(N,))
+
+    # We will need a pdb for MSM building in PyEmma
+    initial_pdb = gro_to_pdb(starting_structure)
+
+    # Get a placeholder object that can serve as a sub context / work graph owner
+    # and can be used in a control operation.
+    subgraph = scalems.subgraph(variables={
+                                    'conformation': initial_input,
+                                    'P': scalems.ndarray(0., shape=(N, N)),
+                                    'is_converged': False})
+
+    with subgraph:
+        modified_input = modify_input(
+            input=initial_input, structure=subgraph.conformation)
+        md = simulate(input=modified_input)
+
+        # Get the output trajectories and pass to PyEmma to build the MSM
+        # Return a stop condition object that can be used in gmx while loop to
+        # terminate the simulation
+        allframes = collect_samples(md.output.trajectory)
+
+        adaptive_msm = analysis.msm_analyzer(topfile=editconf.file['-o'],
```

Comment:
Get a reference to an analyzer that consumes the collection of samples from the current iteration.